server/dns.go

```diff
@@ -78,11 +78,24 @@ func answerQuestion(name string, qtype uint16) []dns.RR {
 
 	// todo: should `shaman.GetRecord` be wildcard aware (*.domain.com) or is this ok
 	// recursively resolve if no records found
-	if len(answers) == 0 {
-		name = stripSubdomain(name)
-		if len(name) > 0 {
-			config.Log.Trace("Checking again with '%v'", name)
-			return answerQuestion(name, qtype)
+	if len(answers) == 0 && []byte(name)[0] != '*' {
```

Comment:
We may want this to not be combined, we still need to be able to check for parent domains if `*` wasn't part of the requested resource.

Reply:
That particular check ensures we don't create an infinite loop. I wasn't able to work out the purpose of responding with a parent domain's details, since that actually seems to contradict the behaviors of other DNS servers.

Reply:
So if somebody can explain why we need/want to return parent domain details...

Reply:
Wow, good thing you pointed this out. So it turns out, we have always supported wildcards, just not properly. Shaman will reply with the parent domain in the answer section, rather than the 'question' domain. Since this PR doesn't address that, it wouldn't resolve the issue.

Reply:
#28 Adds that functionality. Again, great catch. I'd always just `dig @ourdnsip fake.fake.fake.nanoapp.io` which would give an ip, but mismatched answer.

Reply:
To be clear, that's not wildcard support at all, and actively violates the way DNS operates. It works well for use with Nanobox, allowing us to skip creating wildcard subdomains in the first place, but it doesn't actually operate correctly for other use cases at all, as it returns results for domains that _are not defined at all_.

Wildcard support _requires_ a wildcarded entry in the server's records (`*.example.com`) to explicitly state that any value can safely be used to match that part of that entry. What we have currently (after #28 gets merged) works on an assumption that all subdomains should be considered valid, to any arbitrary depth, so long as there is a valid parent domain _somewhere_ in the tree.